lib/Backend/GlobOpt.cpp

```diff
@@ -9611,8 +9621,10 @@ GlobOpt::OptConstFoldUnary(
             }
             else
             {
-                isInt = false;
-                fValue = -(FloatConstType)INT32_MIN;
+                // disable const fold for Math.abs(INT32_MIN) because it causes dst to be float type which
+                // could be different with previous type spec result in LoopPrePass. The latter does int
+                // type spec for Math.abs(INT32_MIN).
+                return false;
```

Comment:
But your change will cause us to type specialize the abs, which will bailout at runtime until we rejit and disable aggressive int type spec.  So it had the same negative cost, plus the bailouts/rejit.
The only other solution would be to not type specialize that case, but then you need to add the lowering to handle the unspecialized abs.  This is a corner case and it doesn't seem worth the cost.